crates/ty_python_semantic/src/place.rs

```diff
@@ -524,6 +524,21 @@ impl<'db> PlaceAndQualifiers<'db> {
         self.qualifiers.contains(TypeQualifiers::CLASS_VAR)
     }
 
+    /// Returns `Some(…)` if the place is qualified with `typing.Final` without a specified type.
+    pub(crate) fn is_bare_final(&self) -> Option<TypeQualifiers> {
+        match self {
+            PlaceAndQualifiers { place, qualifiers }
+                if (qualifiers.contains(TypeQualifiers::FINAL)
+                    && place
+                        .ignore_possibly_unbound()
+                        .is_some_and(|ty| ty.is_unknown())) =>
+            {
+                Some(*qualifiers)
+            }
+            _ => None,
+        }
+    }
```

Comment:
Does this mean that we would consider `x` to be "qualified with bare `Final`" here?

```py
from ty_extensions import Unknown
from typing import Final

x: Final[Unknown]
```

I think that's okay if so; people just shouldn't do that ;)

Reply:
Yes, well spotted :smile:. I agree that it's not as clean as it could be. The alternative is much more complicated as it requires us to return `Option<Type>` and `TypeQualifiers` instead of just `Type` and `TypeQualifiers` from `infer_annotation_expression` and related methods (such that we could return `None` for a bare `Final` or `ClassVar`), which has a lot of downstream consequences.